### PR TITLE
Load top 20 by likes artists/songs in user profile

### PR DESCRIPTION
**Pull request by JekiXD**

### CODE REVIEW

File: frontend/src/app/modules/main/user-profile/user-profile.component.ts [hunk 1]
```diff
@@ -31,8 +31,8 @@ export class UserProfileComponent implements OnInit {
 
   ngOnInit(): void {
     this.getUser();
-    this.getTopArtists();
-    this.loadTopSongs();
+    this.getTop20Artists();
+    this.getTop20Songs();
   }
```
Comment (proskurnicha) [inline frontend/src/app/modules/main/user-profile/user-profile.component.ts:39]: Rename it to getTopArtists, getTopSongs

File: frontend/src/app/modules/main/user-profile/user-profile.component.ts [hunk 2]
```diff
@@ -46,7 +46,13 @@ export class UserProfileComponent implements OnInit {
       });
   }
 
-  getTopArtists = (): Array<object> => new Array<object>();
+  getTop20Artists() {
+    this._artistService.getTopArtistsByLikes(20).subscribe(
+      (result) => {
```
Comment (proskurnicha) [inline frontend/src/app/modules/main/user-profile/user-profile.component.ts:54]: Move to const 20 (magic value)

File: frontend/src/app/modules/main/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -81,19 +87,32 @@ export class UserProfileComponent implements OnInit {
...
+    const gapWidth = parseInt(style.gridGap.split(' ')[0], 10);
+    const elementWidth = parseInt(style.gridTemplateColumns.split(' ')[0], 10);
+
+    return gapWidth + elementWidth;
+  };
+
+  scrollGridRight = (id: string, selector: string) => {
+    const grid = document.getElementById(id);
+
+    grid?.scrollBy({ left: this.getGridScrollWidth(selector) * 3, behavior: 'smooth' });
```
Comment (proskurnicha) [inline frontend/src/app/modules/main/user-profile/user-profile.component.ts]: 3 to const

File: frontend/src/app/modules/main/user-profile/user-profile.component.ts [hunk 4]
```diff
@@ -81,19 +87,32 @@ export class UserProfileComponent implements OnInit {
     element?.scrollBy({ left: this._scrollingSize, behavior: 'smooth' });
   };
 
-  loadTopSongs() {
-    this._songService.getTopSongs().subscribe(
+  getGridScrollWidth = (selector: string) => {
+    const element = document.querySelector(selector);
+    const style = getComputedStyle(element!);
+    const gapWidth = parseInt(style.gridGap.split(' ')[0], 10);
+    const elementWidth = parseInt(style.gridTemplateColumns.split(' ')[0], 10);
```
Comment (proskurnicha) [inline frontend/src/app/modules/main/user-profile/user-profile.component.ts]: 10 to const

**Review by proskurnicha — CHANGES_REQUESTED** [review]

File: backend/ApiGateway/ApiGateway/configuration.Production.json [hunk 5]
```diff
@@ -140,6 +140,19 @@
       "UpstreamPathTemplate": "/api/Songs/file/{id}",
       "UpstreamHttpMethod": [ "GET" ]
     },
+    {
+      "DownstreamPathTemplate": "/api/Songs/top/{amount}",
+      "DownstreamScheme": "https",
```
Comment (NikitaKrasnovSW) [inline backend/ApiGateway/ApiGateway/configuration.Production.json]: Change to http

File: backend/ApiGateway/ApiGateway/configuration.Production.json [hunk 6]
```diff
@@ -192,6 +205,20 @@
         "GET"
       ]
     },
+    {
+      "DownstreamPathTemplate": "/api/Artists/top/{amount}",
+      "DownstreamScheme": "https",
```
Comment (NikitaKrasnovSW) [inline backend/ApiGateway/ApiGateway/configuration.Production.json]: Change to http

**Review by NikitaKrasnovSW — CHANGES_REQUESTED** [review]

**Pull request merged**